hs_swat_modelinstance/receivers.py

```diff
@@ -1,27 +1,48 @@
 from django.dispatch import receiver
+from django.core.exceptions import ObjectDoesNotExist
 
-from hs_core.signals import pre_metadata_element_create, pre_metadata_element_update,pre_create_resource
+from hs_core.signals import pre_metadata_element_create, pre_metadata_element_update,\
+    pre_create_resource, post_metadata_element_update
 
-from hs_swat_modelinstance.models import SWATModelInstanceResource
+import hs_swat_modelinstance.models as swat_models
 from hs_swat_modelinstance.forms import ModelOutputValidationForm, ExecutedByValidationForm,\
     ModelObjectiveValidationForm, SimulationTypeValidationForm, ModelMethodValidationForm,\
     ModelParameterValidationForm, ModelInputValidationForm
 
 
-@receiver(pre_create_resource, sender=SWATModelInstanceResource)
+@receiver(pre_create_resource, sender=swat_models.SWATModelInstanceResource)
 def swatmodelinstance_pre_create_resource(sender, **kwargs):
     metadata = kwargs['metadata']
     modeloutput = {'modeloutput': {'includes_output': False}}
     metadata.append(modeloutput)
 
-@receiver(pre_metadata_element_create, sender=SWATModelInstanceResource)
+
+@receiver(pre_metadata_element_create, sender=swat_models.SWATModelInstanceResource)
 def metadata_element_pre_create_handler(sender, **kwargs):
     return _process_metadata_update_create(**kwargs)
 
-@receiver(pre_metadata_element_update, sender=SWATModelInstanceResource)
+
+@receiver(pre_metadata_element_update, sender=swat_models.SWATModelInstanceResource)
 def metadata_element_pre_update_handler(sender, **kwargs):
     return _process_metadata_update_create(**kwargs)
 
+
+@receiver(post_metadata_element_update, sender=swat_models.SWATModelInstanceResource)
+def check_element_exist(sender, **kwargs):
```

Comment:
This is the only new code in the PR. No there is no unit test that covers this but we could add this later, possibly in a separate issue. Could you point us to code that tests receivers? We don't know how to do that.